pkg/printers/internalversion/describe.go

```diff
@@ -2441,6 +2441,10 @@ func (dd *DeploymentDescriber) Describe(namespace, name string, describerSetting
 			}
 			w.Write(LEVEL_0, "NewReplicaSet:\t%s\n", printReplicaSetsByLabels(newRSs))
 		}
+		overlapWith := d.Annotations[deploymentutil.OverlapAnnotation]
+		if len(overlapWith) > 0 {
+			w.Write(LEVEL_0, "!!!WARNING!!! This deployment has overlapping label selector with deployment %q and won't behave as expected. Please fix it before continuing.\n", overlapWith)
```

Comment:
We can do it atomically in https://github.com/kubernetes/kubernetes/blob/0ed150f3af2b240fd48642e3f0a3a458783b7806/pkg/controller/deployment/sync.go#L262-L275